stubs/SQLAlchemy/sqlalchemy/sql/schema.pyi

```diff
@@ -1,105 +1,205 @@
-    def create(self, bind: Incomplete | None = ..., checkfirst: bool = ...) -> None: ...
-    def drop(self, bind: Incomplete | None = ..., checkfirst: bool = ...) -> None: ...
-    def tometadata(self, metadata, schema=..., referred_schema_fn: Incomplete | None = ..., name: Incomplete | None = ...): ...
-    def to_metadata(self, metadata, schema=..., referred_schema_fn: Incomplete | None = ..., name: Incomplete | None = ...): ...
+    def bind(self) -> Engine | Connection | None: ...
+    def add_is_dependent_on(self, table: Table) -> None: ...
+    def append_column(self, column: ColumnClause, replace_existing: bool = ...) -> None: ...  # type: ignore[override]
+    def append_constraint(self, constraint: Constraint) -> None: ...
+    def exists(self, bind: Engine | Connection | None = ...): ...
+    def create(self, bind: Connectable | None = ..., checkfirst: bool = ...) -> None: ...
+    def drop(self, bind: Connectable | None = ..., checkfirst: bool = ...) -> None: ...
+    def to_metadata(
+        self,
+        metadata: MetaData,
+        schema: str | _symbol | symbol = ...,
+        referred_schema_fn: Callable[[Table, str, ForeignKeyConstraint, str], str] | None = ...,
+        name: str | None = ...,
+    ) -> Table: ...
+    tometadata = to_metadata
 
 class Column(DialectKWArgs, SchemaItem, ColumnClause):
     __visit_name__: str
     inherit_cache: bool
-    key: Any
-    primary_key: Any
-    nullable: Any
-    default: Any
-    server_default: Any
-    server_onupdate: Any
-    index: Any
-    unique: Any
-    system: Any
-    doc: Any
-    onupdate: Any
-    autoincrement: Any
-    constraints: Any
-    foreign_keys: Any
-    comment: Any
-    computed: Any
-    identity: Any
-    info: Any
-    def __init__(self, *args, **kwargs) -> None: ...
-    def references(self, column): ...
-    def append_foreign_key(self, fk) -> None: ...
-    def copy(self, **kw): ...
+    key: str | quoted_name
+    primary_key: bool
+    nullable: bool
+    default: Incomplete | None
+    server_default: Incomplete | None
+    server_onupdate: FetchedValue | str | ClauseElement | None
+    index: int | None
+    unique: bool | None
+    system: bool
+    doc: str | None
+    onupdate: ColumnDefault | Sequence | Incomplete | None
+    autoincrement: bool | str
+    constraints: set[Constraint]
+    foreign_keys: set[ForeignKey]  # type: ignore[assignment]  # supertype is list
+    comment: str | None
+    computed: Incomplete | None
+    identity: Incomplete | None
+    @memoized_property
+    def info(self) -> Mapping[str, Any]: ...
+    # FIXME: Only specifying TraversibleType in __init__ to work around a mypy bug!
+    # example from pandas.tests.io.tests_sql:
+    # ```python
+    # date_type = TEXT if dialect == "sqlite" else DateTime
+    # bool_type = Integer if dialect == "sqlite" else Boolean
+    # Column("DateCol", date_type)
+    # ```
+    # mypy thinks date_type is TraversibleType instead of Union[type[TEXT], type[DateTime]]
```

Comment:
@hauntsaninja This looks like a mypy bug (pyright has no issues with that snippet of code). Are you aware of this? (I haven't tried in 1.0 either) All those types should be valid `type[TypeEngine]`, but instead mypy narrows it down to their metaclass `TraversibleType`.

mypy_primer comment before I added this workaround: https://github.com/Avasam/typeshed/pull/24#issuecomment-1407323271
https://github.com/Avasam/typeshed/actions/runs/4030841677/jobs/6929764991